http/server.go

```diff
@@ -402,3 +523,41 @@ func (h *handler) Health(w http.ResponseWriter, r *http.Request) {
 	b, _ = json.Marshal(resp)
 	_, _ = w.Write(b)
 }
+
+func readChainHash(r *http.Request) ([]byte, error) {
+	var err error
+	chainHashHex := make([]byte, 0)
+
+	chainHash := chi.URLParam(r, chainHashParamKey)
+	if chainHash != "" {
+		chainHashHex, err = hex.DecodeString(chainHash)
+		if err != nil {
+			return nil, fmt.Errorf("invalid chain hash")
+		}
+	}
+
+	return chainHashHex, nil
+}
+
+func readRound(r *http.Request) (uint64, error) {
+	round := chi.URLParam(r, roundParamKey)
+	return strconv.ParseUint(round, roundNumBase, roundNumSize)
+}
+
+func (h *DrandHandler) getBeaconHandler(chainHash []byte) (*beaconHandler, error) {
+	chainHashStr := fmt.Sprintf("%x", chainHash)
+	if chainHashStr == "" {
+		chainHashStr = common.DefaultChainHash
+	}
+
+	h.state.Lock()
```

Comment:
This is going to get called on every inbound http client request.
This structure is going to have lock contention.
It at least needs to be an `RWMutex` because you are only `Rlock` ing it here, not needing to acquire an exclusive lock.

Reply:
ok, fixed it